components/automate-ui/src/app/pages/team-details/team-details.component.ts

```diff
@@ -107,6 +120,30 @@ export class TeamDetailsComponent implements OnInit, OnDestroy {
         // goes to #users if (1) explicit #users, (2) no fragment, or (3) invalid fragment
         this.tabValue = (fragment === 'details') ? 'details' : 'users';
       });
+
+    this.store.select(iamMinorVersion)
+      .pipe(takeUntil(this.isDestroyed))
+      .subscribe((version) => {
+        if (version === null) { return; }
+        this.isMinorV1 = version === 'v1';
+      });
+
+    this.store.select(assignableProjects)
+      .subscribe((assignable: ProjectsFilterOption[]) => {
+        assignable.forEach(p => {
+          const proj = <Project>{
```

Comment:
I was casting because we do need to cast from `string` to `IAMType` but I can make that a bit more explicitly with:


```
          const proj: Project = {
            id: p.value,
            name: p.label,
            type: <IAMType>p.type
          };
```

Reply:
We don't have to cast that, we could also do what we'd do in golang -- _type guards_. see here for docs: https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

in terms of effort, it's bigger than the case here, though.